packages/components/src/query-controls/index.js

```diff
@@ -76,16 +84,20 @@ export default function QueryControls( {
 			/>
 		),
 		onCategoryChange && (
-			<CategorySelect
-				key="query-controls-category-select"
-				categoriesList={ categoriesList }
-				label={ __( 'Category' ) }
-				noOptionLabel={ __( 'All' ) }
-				selectedCategoryId={ selectedCategoryId }
+			<FormTokenField
+				label={ __( 'Categories' ) }
+				value={
+					selectedCategories &&
+					selectedCategories.map( ( item ) => ( {
+						id: item.id,
+						value: item.name || item.value,
+					} ) )
+				}
+				suggestions={ Object.keys( suggestions ) }
```

Comment:
the `FormTokenField` has a `maxSuggestions` prop what we should use for cases when categories are not for content and so are too many in number.

Reply:
What value should be max?

Reply:
Let's use a constant and the same value the `FlatTermSelector` component [is using](https://github.com/WordPress/gutenberg/blob/a481717772f783a29a80d682141b4da62bc8d5a2/packages/editor/src/components/post-taxonomies/flat-term-selector.js#L36):
```
const MAX_TERMS_SUGGESTIONS = 20;
```